source/leop.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2015  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/**
+ * @file leop.c
+ * @author Grace Yi
+ * @date Oct. 2021
+ */
+
+#include "../include/leop.h"
+
+Deployable_TypeDef sw;
+
+//LEOP Sequence
+
+/**
+ * @brief
+ *      Check hardswitch status to ensure all deployables 
+ *      have been successfully deployed
+ * @return bool
+ *      Returns TRUE if all deployables have been deployed
+ *      Returns FALSE otherwise
+ */
+bool hard_switch_status() {
+    int getStatus_retries;
+    int successful_deployment = 0;
+    //int i = 0;
+    //char hard_switches[8] = {Port, UHF_P, UHF_Z, Payload, UHF_S, UHF_N, Starboard, DFGM};
+    for (getStatus_retries = 1; getStatus_retries <= 3; getStatus_retries++) {
+        for (sw = 0; sw < 8; sw++) {
```

Comment:
Is there a macro or definitions somewhere for the number of switches? If not let's add one so that if we add a device to this sequence we don't have to hunt around for "8"